src/robusta/core/model/base_params.py

```diff
@@ -70,6 +71,45 @@ def post_initialization(self):
     pass
 
 
+class ResourceInfo(BaseModel):
+    name: str
+    namespace: Optional[str]
+    kind: str
+    node: Optional[str]
+    container: Optional[str]
+
+
+class HolmesParams(ActionParams):
+
+    holmes_url: Optional[str]
+
+    @validator("holmes_url", allow_reuse=True)
+    def validate_protocol(cls, v):
+        if v and not v.startswith("http"):  # if the user configured url without http(s)
+            v = f"http://{v}"
+            logging.info(f"Adding protocol to holmes_url: {v}")
+        return v
+
+
+class AIInvestigateParams(HolmesParams):
+    """
+    :var resource: The resource related to this investigation. A resource has a `name` and `kind`, and may have `namespace` and `node`
+    :var investigation_type: The type of investigation: Issue/Service/Cluster/Custom
+    :var runbooks: List of human readable recommended runbooks that holmes can use for the investigation.
```

Comment:
we support custom runbooks from the action? cool

Reply:
We didn't implement it yet, but it's coming very soon, and it's going to be super cool!